src/chmfile.cpp

```diff
@@ -526,8 +526,10 @@ bool CHMFile::BinaryIndex(CHMListCtrl& toBuild, const wxCSConv& cv)
                 }
             }
 
-            if (bt_ui.length < offset + 8)
+            if (bt_ui.length < offset + 8) {
+                toBuild.UpdateItemCount();
```

Comment:
On this patch, did you happen to profile xCHM with and without the `UpdateItemCount()` changes? Is a `SetItemCount()` call really expensive (I ask because the wxWidgets documentation states that no repainting happens when you call it: https://docs.wxwidgets.org/3.0/classwx_list_ctrl.html#a5ded9342e4969eeeb60d1305f75995fd).

In any case, if you did profile it and this really helped: there are two `UpdateItemCount()` calls, but there are more places in `BinaryIndex()` where `item` could be `!= 0` and we return - and we should, by that measure, call `UpdateItemCount()` there too (for example, at lines 496, 507, 517). But. Rather than doing that, we should rather be consistent with your other strategy from chmsearchpanel.cpp, and just do:

```
    toBuild.Freeze();
    bool bindex = BinaryIndex(toBuild, *cvPtr);
    toBuild.UpdateItemCount(); 
    toBuild.Thaw();
```

Reply:
Done. And yes I profiled it. It's more than twice as fast now.

patch:

```
CHMApp::OnInit	0.00s	3.71s	0.00%	22.21%	xchm	C:/src/xCHM/src/chmapp.cpp	384	0x7ff6239f2e64
```
original:
```
CHMApp::OnInit	0.00s	10.74s	0.00%	24.00%	xchm	C:/src/xCHM/src/chmapp.cpp	384	0x7ff7d1792e64
```